.golangci.yml

```diff
@@ -1,23 +1,165 @@
+  # comment: godot, godox, misspell
+  # complexity: cyclop, funlen, gocognit, gocyclo, nestif
+  # error: errcheck, errorlint, goerr113, wrapcheck
+  # format: gci, gofmt, gofumpt, goimports
+  # import: depguard, gci, goimports, gomodguard
+  # metalinter: gocritic, govet, revive, staticcheck
+  # module: depguard, gomoddirectives, gomodguard
+  # performance: bodyclose, maligned, noctx, prealloc
+  # sql: rowserrcheck, sqlclosecheck
+  # style: asciicheck, depguard, dogsled, dupl, exhaustivestruct, forbidigo, forcetypeassert, gochecknoglobals, gochecknoinits, goconst, gocritic, godot, godox, goerr113, goheader, golint, gomnd, gomoddirectives, gomodguard, goprintffuncname, gosimple, ifshort, importas, interfacer, lll, makezero, misspell, nakedret, nlreturn, nolintlint, paralleltest, predeclared, promlinter, revive, stylecheck, tagliatelle, testpackage, thelper, tparallel, unconvert, wastedassign, whitespace, wrapcheck, wsl
+  # test: exhaustivestruct, paralleltest, testpackage, tparallel
+  # unused: deadcode, ineffassign, structcheck, unparam, unused, varcheck
+
+  enable: []
+  # Enabled by default linters:
+  # deadcode: Finds unused code [fast: false, auto-fix: false]
+  # errcheck: Errcheck is a program for checking for unchecked errors in go programs. These unchecked errors can be critical bugs in some cases [fast: false, auto-fix: false]
+  # gosimple (megacheck): Linter for Go source code that specializes in simplifying a code [fast: false, auto-fix: false]
+  # govet (vet, vetshadow): Vet examines Go source code and reports suspicious constructs, such as Printf calls whose arguments do not align with the format string [fast: false, auto-fix: false]
+  # ineffassign: Detects when assignments to existing variables are not used [fast: true, auto-fix: false]
+  # staticcheck (megacheck): Staticcheck is a go vet on steroids, applying a ton of static analysis checks [fast: false, auto-fix: false]
+  # structcheck: Finds unused struct fields [fast: false, auto-fix: false]
+  # typecheck: Like the front-end of a Go compiler, parses and type-checks Go code [fast: false, auto-fix: false]
+  # unused (megacheck): Checks Go code for unused constants, variables, functions and types [fast: false, auto-fix: false]
+  # varcheck: Finds unused global variables and constants [fast: false, auto-fix: false]
+
+  disable:
+    - deadcode # deprecated
+    - exhaustivestruct # deprecated
+    - golint # deprecated
+    - ifshort #deprecated
+    - interfacer # deprecated
+    - maligned # deprecated
+    - nosnakecase # depreceated
+    - scopelint # deprecated
+    - structcheck # deprecated
+    - varcheck # deprecated
+    # TODO: handle the issue with the following linters
+    - gochecknoglobals # flags our globals
+    - depguard # flags our modules
+    - forbidigo # flags our fmt.Println
+    - wrapcheck # flags our unchecked errors
```

Comment:
**thought:** I'm a bit worried about generating too much noise from linters, this is why I went with “whitelist linters” approach, but you seem to be reverting to “blacklist linters” approach here.

For me, biggest priority was from message that "errcheck" and "gocritic" raised.

I'm open to hearing your thoughts on this – it's my first year writing golang, so I don't have strong opinions of my own.


I normally do what "charm" team does and here is how they approach go linting:
https://github.com/charmbracelet/glow/blob/master/.golangci.yml